azurerm/cache.go

```diff
@@ -0,0 +1,45 @@
+package azurerm
+
+import (
+	"context"
+
+	"github.com/cycloidio/terracognita/errcode"
+	"github.com/cycloidio/terracognita/provider"
+	"github.com/cycloidio/terracognita/tag"
+	"github.com/pkg/errors"
+)
+
+func cacheVirtualNetworks(ctx context.Context, a *azurerm, rt string, tags []tag.Tag) ([]provider.Resource, error) {
+	rs, err := a.cache.Get(rt)
+	if err != nil {
+		if errors.Cause(err) != errcode.ErrCacheKeyNotFound {
+			return nil, errors.WithStack(err)
+		}
+
+		rs, err = virtualNetworks(ctx, a, rt, tags)
```

Comment:
it's a bit redundant as the function in the reader already wraps the error saying mostly the same thing.

Reply:
yep, but it adds more context: "you're unable to get virtual networks because you have an error with the reader or an error with the fact you can't set the name of the resource data."

instead of having directly the reader error / the name error without more information. You actually add an additional context layer :)